Piranha Plant Rework (Redux): Adaptive Roots

H3ra-57 opened the pull request:
This rework aims to turn Piranha Plant into a stance character with a new mechanic, Adaptive Roots. The standard plant has the best movement, the putrid plant is the most defensive with attacks capable of poisoning the foe, and the prickly plant packs the biggest punch with heavy, yet slow attacks.

### Entering Stances
- Simply taunt to change stance. Up taunt changes to the standard plant, down taunt changes to Putrid Plant, and side taunt changes to Prickly Plant.
### General Stance Changes
- Plant moves slower when Putrid/Prickly
- Putrid's attacks deal less damage, and bites deal poison DoT
- Prickly's attacks are slowed down, but deal more damage, head attacks most of all
### Ptooie
- Putrid Ptooie deals poison DoT
- Prickly Ptooie is larger and deals more damage, but goes a shorter distance
### Poison Breath → Adaptive Maw
Poison Cloud has been changed into a completely new move, Adaptive Maw- a close-range spit that has different properties depending on Plant's current stance!
- Standard Plant spits a gust of wind that sends at a nasty angle
- Putrid Plant spits a multihit poison breath that poisons
- Prickly Plant bites instead of spitting, dealing high shield damage and piercing shields at full charge
### Piranhacopter
- Is now a single hit
- Grounded version no longer leaves the ground
- Cancel goes into special fall immediately
- Aerial version moves much slower
### Changelog and hitbox diffs under construction
<<<=== PIRANHA PLANT===>>>
STATS/ADAPTIVE ROOTS:
~ (!) Can now change stance by pressing taunt. Up taunt activates standard plant, side taunt activates prickly plant, and down taunt activates putrid plant.
~~ Putrid:
~ (!) Moves at 88% speed
~ (!) Head attacks have 60% power, bite attacks have 85% power, and other attacks have 70% power
~ (!) Bite attacks deal poison DoT damage
~~ Prickly:
~ (!) Moves at 75% speed
~ (!) Head attacks have 1.4x power, bite attacks have 1.05x power, and other attacks have 1.2x power
~ (!) Attacks have increased startup

Jab [Other]:
~~ Jab 1:
~ [+] Hitbox Size (ID0/1/2&3): 1.8u/1.8u/2.0u → 2.5u/2.5u/3.0u
~~ Jab 2:
~ [+] Hitbox Size (ID0/1/2&3): 2.6u/2.8u/3.0u → 3.2u/3.4u/4.0u
~ (!) Prickly cannot do rapid jab
~~ Jab 3 [Head]:
~~~ Prickly:
~ Startup: F10
~~ Rapid Jab Finisher [Bite]:
~~~ Putrid:
~ Poison DoT: 2%

Forward Tilt [Bite]:
~~ Hit 1:
~~~ Prickly:
~ Startup: F10
~~ Hit 2:
~~~ Putrid:
~ Poison DoT: 3.0%
~~~ Prickly:
~ Startup: F8

Up Tilt [Head]:
~~~ Prickly:
~ Startup: F11

Down Tilt [Other]:
~ [-] Startup: → F8
~~~ Putrid:
~ Knockback compensated
~~~ Prickly:
~ Startup: F11

Dash Attack [Other]:
~ [-] Hitbox Duration: F7-F22 -> F7-F19
~~ Late Hit: 
~ [-] Angle: 85 → 50 
~~~ Prickly:
~ Startup: F11

Forward Smash [Head]:
~ [-] Endlag increased (F63? motion rate removed on endlag)
~~~ Prickly:
~ Startup: F26

Up Smash [Bite]:
~~~ Putrid:
~ Poison DoT: 4.0%
~~~ Prickly:
~ Startup: F16

Down Smash [Other]:
~ [-] Startup: → F14
~~~ Prickly:
~ Startup: F19

Neutral Air [Other]:
~~ Multihits: 
~ [-] Damage: 2.5% → 2% 
~ [-] Angle: 367 → 365
~~~ Prickly:
~ Startup: F8

Forward Air [Other]:
~ [-] Intangibility removed
~ [-] Startup: → F9
~ [+] Interpolation added
~~~ Prickly:
~ Startup: F13

Back Air [Other]:
~ [/] Angle: 361 → 50
~~~ Prickly:
~ Startup: F18
~~~ Putrid:
~ Reworked into a multihit
~~ Multihit (F14 → F25)
~ Size: 7.0u
~ Damage: 1.4%
~ Angle: 366
~ KBG: 100
~ BKB: 15
~~ Launcher (F26 → F30)
~ Size: 9.0u
~ Damage: 4.9%
~ Angle: 50
~ KBG: 108
~ BKB: 25
~ Poison DoT: 3.0%

Up Air [Head]:
~ [-] Intangibility removed
~ [-] Startup: F6 → F9
~~~ Putrid:
~ Knockback compensated
~ Angle: 85 → 60 
~~~ Prickly:
~ Startup: F11

Down Air [Other]:
~ [-] Startup: F9 → F11
~ [/] KBG (ID3) fixed: 90 → 64
~~~ Prickly:
~ Startup: F14

Forward Throw [Head]:
~~~ Prickly:
~ Startup: F17

Down Throw [Bite]:
~~~ Putrid:
~ Poison DoT: 0.5%

Ptooie:
~ [-] Damage (Weak/Strong): 14%/18% → 12%/15%
~~~ Putrid:
~ Damage (Weak/Strong): 10%/12%
~ Poison DoT: 4%
~ Does not bounce
~~~ Prickly:
~ Damage & Size increased by 1.3x
~ Shoot distance reduced

Poison Breath → Adaptive Maw:
~ (!) Now a close-range spit that changes properties depending on the stance
~ Hitbox Size: 9.0u
~~~ Standard:
~ Hitbox Duration: F9-14
~ Damage (Uncharged/Charged): 8%/12%
~ BKB: 60
~ KBG: 66
~ Angle: 30
~~~ Putrid:
~~ Multihits:
~ Hitbox Duration: F8-24
~ Damage (Uncharged/Charged): 1.5%/2.5%
~ BKB: 10
~ KBG: 66
~ Angle: 366
~~ Finisher:
~ Hitbox Duration: F24-27
~ Damage (Uncharged/Charged): 3%/2%
~ BKB: 70
~ KBG: 66
~ Angle: 50
~ Poison DoT (Uncharged/Charged): 3.0%/5.0%
~~~ Prickly:
~ Hitbox Duration: F13-16
~ Damage (Uncharged/Charged): 10%/16%
~ FKB: 30
~ KBG: 100
~ Angle: 290
~ 5.0 shield damage uncharged, ignores shield at full charge

Piranhacopter:
~ [-] Startup: → F10
~ [-] Now remains grounded when used on the ground
~ [-] Vertical speed: 1.7 → 1.0
~ [-] Horizontal speed: 2.4 → 1.4
~ [R] Aerial cancel now transitions straight to special fall
~ [R] Multihits removed
New hitbox data:
Strong hit (F10 → F33)
~ Damage: 10%
~ Angle: 30°/40° (grounded/aerial)
~ BKB: 70/50 (grounded/aerial)
~ KBG: 70
~ Hitlag: 1.0
~ Hitbox size: 3.5/3.5/5.0/5.0 (ID 0/1/2/3)
~ Hitbox X Position: 6.0/6.0/1.5/1.5 (ID 0/1/2/3)
Late hit (F33 → F65)
~ Damage: 7%
~ Angle: 40°/50° (grounded/aerial)
~ BKB: 70/50 (grounded/aerial)
~ KBG: 70
~ Hitlag: 1.0
~ Hitbox size: 3.5/3.5/5.0/5.0 (ID 0/1/2/3)
~ Hitbox X Position: 6.0/6.0/1.5/1.5 (ID 0/1/2/3)

Long-Stem Strike:
~~~ Putrid:
~ Poison DoT (Uncharged/Charged): 2.0%/3.0%

h3ra-fa2019 committed:
rudimentary stance status baseline

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

h3ra-fa2019 committed:
placeholder anim

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

h3ra-fa2019 committed:
stance properly sets speeds; new side b; start of changed neutral b

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

h3ra-fa2019 committed:
side b hitbox adjustment; neutral b; setup for normals

H3ra-57 committed:
Merge branch 'plant-side-b' into dev_to_plant

H3ra-57 committed:
Merge pull request #2 from h3ra-ai2020/dev_to_plant

uaggh

h3ra-fa2019 committed:
stance cleanup; normals

h3ra-fa2019 committed:
putrid side b fix

h3ra-fa2019 committed:
RECOVERY SWEEP (this commit needs to be put into the recovery sweep)

h3ra-fa2019 committed:
light poison side b poisons too

h3ra-fa2019 committed:
prickly side b a little less laggy

H3ra-57 committed:
Merge branch 'dev' into plant-side-b

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

H3ra-57 committed:
normalize angles

H3ra-57 committed:
normalize hitlag/shield damage

Detronlegacy commented:
Wavedashes allow plant to ignore speed reductions caused by stance switching.

WoodrowManuel commented:
Putrid Ptooie's initial hitbox was moved from the lips to the base of the pot.
https://streamable.com/5cnggo

H3ra-57 committed:
Merge remote-tracking branch 'upstream/dev' into plant-side-b

H3ra-57 committed:
Merge remote-tracking branch 'upstream/dev' into plant-side-b

H3ra-57 committed:
stance change to taunts, other adjustments

H3ra-57 committed:
nair hitboxes unduplicated

H3ra-57 committed:
uthrow does not change; ftilt bugfix

H3ra-57 committed:
Merge remote-tracking branch 'upstream/dev' into plant-side-b

H3ra-57 committed:
normalize putrid side b numbers

H3ra-57 committed:
putrid down b poisons; starting on askew's recommended changes

H3ra-57 committed:
Merge remote-tracking branch 'upstream/dev' into plant-side-b

H3ra-57 committed:
typo in jab2

H3ra-57 committed:
Merge branch 'plant-side-b' of https://github.com/h3ra-ai2020/HewDraw-Remix into plant-side-b

H3ra-57 committed:
Merge remote-tracking branch 'upstream/dev' into plant-side-b

H3ra-57 committed:
Merge branch 'plant-side-b' of https://github.com/h3ra-ai2020/HewDraw-Remix into plant-side-b

H3ra-57 committed:
putrid is even more stinky

H3ra-57 committed:
more poison damage for ftilt

H3ra-57 committed:
up b damage changed per stance

jobrien97 committed:
upB initial touchups

SkewedAskew committed:
Big batch of tweaks

SkewedAskew committed:
More adjustments

- Reduced non-Prickly nair multihit frame advantage
- Moved Putrid bair hitbox inward and refined effects further
- Adjusted Putrid bair multihit frame advantage
- Prickly Ptooie startup F8 -> F11

SkewedAskew committed:
Even More Adjustments

- Cleaned up many vfx and added/adjusted sfx
- Normalized jab2 knockback to mitigate Prickly jab3 not connecting
- Putrid pummel damage 1.4% -> 1.7%

SkewedAskew committed:
Update opff.rs

Allows stance changing while on angel platform

SkewedAskew committed:
Merge branch 'dev' into pr/1379

SkewedAskew committed:
Motion rate cleanup

Moves FT_MOTION_RATE calls out of is_excute blocks

SkewedAskew committed:
Update aerials.rs

SkewedAskew committed:
Crash fix?

jobrien97 commented on fighters/packun/src/opff.rs:
```diff
@@ -4,22 +4,153 @@ use super::*;
+                y: 0.0, 
+                z: 0.0
+            };
+            KineticModule::add_speed_outside(fighter.boma(), *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
```
Please don't use wind, dash should already be covered in the control energy status speed mul, no?

jobrien97 commented on fighters/packun/src/opff.rs:
```diff
@@ -4,22 +4,153 @@ use super::*;
+/// This should only be called once per status, or you will get some multiplicative effects
+unsafe fn apply_status_speed_mul(fighter: &mut smash::lua2cpp::L2CFighterCommon, mul: f32) {
+    // set the X motion speed multiplier (where movement is baked into an anim)
+    lua_bind::FighterKineticEnergyMotion::set_speed_mul(fighter.get_motion_energy(), mul);
```
Please do
```
fighter.clear_lua_stack();
lua_args!(fighter, FIGHTER_KINETIC_ENERGY_ID_MOTION);
let og_speed_mul = app::sv_kinetic_energy::get_speed_mul(fighter.lua_state_agent);
lua_bind::FighterKineticEnergyMotion::set_speed_mul(fighter.get_motion_energy(), og_speed_mul * mul);
```
instead. This allows dash attack to be naturally affected by your status speed mul.

jobrien97 commented on fighters/packun/src/acmd/ground.rs:
```diff
@@ -65,45 +75,78 @@ unsafe fn packun_attack_12_game(fighter: &mut L2CAgentBase) {
+    let stance = StanceInfo::from(VarModule::get_int(boma.object(), vars::packun::instance::CURRENT_STANCE));
     // base 0.93
-    sv_kinetic_energy!(set_speed_mul, fighter, FIGHTER_KINETIC_ENERGY_ID_MOTION, 0.8);
+    sv_kinetic_energy!(set_speed_mul, fighter, FIGHTER_KINETIC_ENERGY_ID_MOTION, stance.da_speed);
```
Please revert this in favor of what I commented in opff.rs

SkewedAskew committed:
Merge branch 'dev' into pr/1379

SkewedAskew committed:
Final adjustments

- Made fsmash neck a sourspot instead of accidentally being stronger, heavily increased Putrid sweetspot kb to be usable
- Small startup decreases to certain Prickly moves (utilt/smashes/side b)
- Rebalanced uthrow to be best at killing in Putrid stance
- Minor improvements to up b speed and maneuverability

SkewedAskew committed:
Merge branch 'dev' into pr/1379

SkewedAskew committed:
Nair frame advantage

- Also cleans up Prickly side b a bit more

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

SkewedAskew committed:
Vfx/sfx tweaks, speed_mul fixes

SkewedAskew committed:
Merge branch 'plant-side-b' of https://github.com/h3ra-ai2020/HewDraw-Remix into pr/1379

SkewedAskew committed:
Merge branch 'dev' into pr/1379

SkewedAskew committed:
Speed_mul fixes

- Removes speed_mul on all rolls and fixes dash to run energy

SkewedAskew committed:
Merge branch 'dev' into pr/1379

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into plant-side-b

SkewedAskew committed:
Merge branch 'dev' into pr/1379

SkewedAskew committed:
Merge branch 'plant-side-b' of https://github.com/h3ra-ai2020/HewDraw-Remix into pr/1379

SkewedAskew committed:
Small cleanup

H3ra-57 committed:
Merge branch 'dev' into plant-side-b
